internal/context/prompts.go

```diff
@@ -147,3 +149,43 @@ func PromptModuleParams(moduleConfig moduleconfig.ModuleConfig, parameters map[s
 	}
 	return parameters, nil
 }
+
+func promptCredentialsAndFillProjectCreds(credentialPrompts map[string][]PromptHandler, credentials globalconfig.ProjectCredential) globalconfig.ProjectCredential {
+	promptsValues := map[string]map[string]string{}
+
+	for vendor, prompts := range credentialPrompts {
+		vendorPromptValues := map[string]string{}
+
+		// vendors like AWS have multiple prompts (accessKeyId and secretAccessKey)
+		for _, prompt := range prompts {
+			vendorPromptValues[prompt.Field] = prompt.GetParam(map[string]string{})
+		}
+		promptsValues[vendor] = vendorPromptValues
+	}
+
+	// FIXME: what is a good way to dynamically modify partial data of a struct
+	// current just marashing to yaml, then unmarshaling into the base struct
+	yamlContent, _ := yaml.Marshal(promptsValues)
+	yaml.Unmarshal(yamlContent, &credentials)
+
+	globalconfig.Save(credentials)
```

Comment:
The name of this function doesn't make it very clear that it's actually going to save the file. Maybe rename or just return the credentials and let the caller save it?